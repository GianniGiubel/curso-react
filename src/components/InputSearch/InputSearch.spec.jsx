import { render,screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { InputSearch } from '.';

describe('<InputSearch/>',() => {
  it('should have a value on searchValue', () => {
    const fn = jest.fn();
    render(<InputSearch handleChange = {fn} searchValue={'Testando'}></InputSearch>);
    
    const input = screen.getByPlaceholderText(/search/i);
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('Testando');
  }); 

  it('should call HandleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<InputSearch handleChange = {fn}></InputSearch>);
    
    const input = screen.getByPlaceholderText(/search/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  }); 

  it('should match snapshot', () => {
    const fn = jest.fn();
    const {container} = render(<InputSearch handleChange = {fn}></InputSearch>); 

    expect(container.firstChild).toMatchSnapshot();
  });  
});