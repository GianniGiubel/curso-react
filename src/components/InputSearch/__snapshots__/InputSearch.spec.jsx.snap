// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch/> should match snapshot 1`] = `
<input
  class="inputSearch"
  placeholder="Search"
  type="search"
  value=""
/>
`;
